### DISCUSSION TITLE
How to set custom circuit parameters in code?

### DISCUSSION THREAD

**Discussion by YulienPohl** [hneemann/Digital#674] [Q&A]
According to the help menu, one should set the parameters like this.
```java
circ := addComponent("circ.dig", 0, 0);
circ.generic = "bits := 8"; // does not work
circ.bits := 8; // works with := not =

circ.rotation = 1;   // does not work;
circ.Label = "text"; // does not work;
```

And how to set the intFormat?

**Comment by hneemann** [comment]
You have to distinguish two things when embedding a generic circuit. One is the custom component. This cannot be changed by the code in the custom component. And the other is the actual embedded circuit itself, which usually requires parameters. These parameters can be defined. 

This is quite different from simple components. In the simple components, the generic code can change the component itself. The generic code of a user defined component, can only parameterize the included circuit.

**Reply by YulienPohl** [reply]
But then why is (Name to use in generic circuits: rotation) in the custom circuit info menu, when you can't rotate it with code?

**Reply by hneemann** [reply]
This is because the help window is generated in a generic way and in other components it is possible to set the rotation.

**Reply by YulienPohl** [reply]
Okay, that means there is currently no other way except to set it manually.

**Reply by hneemann** [reply]
Thinking about it, I realized that this can be implemented very easily.

So in head this is now possible with 

```
a:=addComponent("a.dig",x,y);
a.this.rotation=2;
```
to change the component itself.
